### ISSUE TITLE
start4.elf is icompatible with up to date PI4 board

### ISSUE THREAD

**Issue by BruhP8** [drakkar-lig/walt-rpi-debian#1]
Hello
When trying to run the latest release of WalT on a brand new PI4 2GB board, the bootloader hangs : 
```
bootloader: 6efe41bd Jul 6 2021 
....
Read config.txt bytes               32 hnd 0x00000000
Read  start4.elf bytes     2277376 hnd 0x00000000
Read fixup4.dat bytes          5407 hnd 0x00000000
Firmware: 21a15cb094f41c7506ad65d2cb9b29c550693057 Jul 17 2020 10:59:17 0x00b03115 0x00000000 0x0000001f
start4.elf: is not compatible
This board requires newer software
```
I have tried updating the board firware, bios and bootloader to no avail. After doing some research, it looks like the board's bootloader is not compatible with the WalT release.
RPI os boots just fine.
It would appear that current revisions of the PI4B board (1.5 at the time of writing, marked b03115 in the board section when in the bootloader) are not compatible with WalT's release of the start4.elf file.
Is anyone aware of the issue ?

Edit : I appear to not be alone : since April 2021, when rev1.5 was released *something* has changed. See : https://www.youtube.com/watch?v=wvZPg9eMlQo 
for an example. 
TLDR seem to be that rev 1.5 changed the boot method used, without taking back compatibility in consideration

Thanks

**Comment by eduble** [comment]
I also met this issue recently.

Could you try the files from this archive instead, on your SD card?
[rpi-sd-files.tar.gz](https://github.com/drakkar-lig/walt-rpi-debian/files/8560810/rpi-sd-files.tar.gz)
This archive should be part of walt version 8, which is not ready yet; so I will let this issue open until the release is published.

In this archive, there are files allowing to update the EEPROM on the rpi4 and set the boot order to:
1. network boot (firmware-based, allowing the board to boot without a SD card)
2. sd card boot
3. reset the board and restart to 1

The reason for this is that sd card boot (firmware -> u-boot -> kernel) has minor issues with recent rpi4 boards (SD card reader not accessible once the OS is booted), whilst the network boot (firmware -> kernel) works perfectly.
Note: we still need to boot once from the sd-card at the very first time, for proper registration of the walt node on the walt server.
Once the EEPROM is updated (and boot order is updated), the firmware automatically renames file `recovery.bin` to `RECOVERY.000` in order to prevent this EEPROM update to be run again and again at each reboot.
However, if you want to use the same SD card for the first boot of several rpi4 boards, you will have to rename `RECOVERY.000` back to `recovery.bin` to re-enable the EEPROM update after each boot.

By the way, can you tell me for which company or research lab you are working?

**Comment by BruhP8** [comment]
Hello eduble
Using your files did make the node boot properly, thanks !
I find it mind blowing that they would change something like this on a minor board revision, which is only supposed to be hardware changes. I did some more research after posting, and found quite a bit of people complaining about this issue, and how it breaks a lot of pi based distros other than RpiOS.

Anyway, thanks for your time. 
I am an intern at Avisto

**Comment by eduble** [comment]
OK.
Anyway, could you make one more check: does the node still boot, if you reboot it once more, now that the EEPROM selects network boot first?
In the network boot case, the files usually found on the SD card are instead requested using TFTP, thus the node actually gets the firmware files at `[image]:/boot/rpi-4-b/<file>`.
So I suppose that with an old image (with obsolete firmware files there) you would still get issues with this network boot.
If you don't notice this problem, then it probably means the firmware uses sd card boot as its second boot order option (after detecting the firmware files obtained from TFTP are obsolete).
Did you try booting with the SD card unplugged, to be sure that we are using the network boot?
I have built an updated walt default image for rpi4 with embedded firmware files updated. Network boot works good with it. If you need it, I can push it on the docker hub.
